metrics/config.go

```diff
@@ -341,6 +353,25 @@ func prometheusPort() (int, error) {
 	return int(pp), nil
 }
 
+// prometheusHost returns the host configured via the environment
+// for the Prometheus metrics exporter if it's set, a default value otherwise.
+// Validation is performed via parsing the host value as part of a url.
+func prometheusHost() (*string, error) {
+	phStr := os.Getenv(prometheusHostEnvName)
+	// empty string either explicitly set via the env var or by default
+	// if the env var is not set translates to "0.0.0.0"
+	if phStr == "" {
+		return &phStr, nil
```

Comment:
Shouldn't this return "0.0.0.0" instead of empty?

Reply:
This is what http uses as a convention and it is commonly used, returning `0.0.0.0` will add more changes to this PR for the sake of maybe clarity.

Reply:
Yes, I realized that after reading the whole code.